auditlog_security/models/auditlog_rule.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2021 Therp B.V.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import exceptions, models, fields, api, modules, _
+from odoo.addons.auditlog.models.rule import FIELDS_BLACKLIST
+from odoo import SUPERUSER_ID
+
+
+class AuditlogRule(models.Model):
+    _inherit = "auditlog.rule"
+
+    auditlog_line_access_rule_ids = fields.One2many(
+        "auditlog.line.access.rule", "auditlog_rule_id", ondelete="cascade"
+    )
+    server_action_id = fields.Many2one('ir.actions.server', "Server Action")
+
+
+    def sudo(self, user=SUPERUSER_ID):
+       return super(
+            AuditlogRule, self.with_context(real_user=self.env.context.get('uid'))
+            ).sudo(user=user)
+
+    @api.onchange("model_id")
+    def onchange_model_id(self):
+        # if model changes we must wipe out all field ids
+        self.auditlog_line_access_rule_ids.unlink()
+
+    @api.multi
+    def unlink(self):
+        lines = self.mapped("auditlog_line_access_rule_ids")
+        res = super(AuditlogRule, self).unlink()
+        if res:
+            lines.unlink()
+        return res
+
+    @api.multi
+    def _get_view_log_lines_action(self):
+        self.ensure_one()
+        fields_ids = self.auditlog_line_access_rule_ids.mapped('field_ids').ids
```

Comment:
Actually, this makes no sense. Why should we include the `field_ids` in the domain to generate the view log lines action? It should not have a domain, because you should see all the lines you are allowed to see, which is governed by Odoo's `ir.rule` records, not by these `auditlog.rule` records.

Now, we limit ourselves because this function is just called for one rule only, and it will tailor the domain to show the lines based on that rule only, but if a user is added to multiple rules, it will not work. Also, when the rule is deleted, the button will stop working, whereas an admin user should still be able to use this function and click it and just see all lines.

Conclusion:

1. This function should become an `@api.model` and we should not pass the rule to it anymore
2. We should not include the `fields` in the domain, just the model.
3. The model should be passed to this function, or this function should be moved to `ir.model` itself.

Reply:
Agreed the records will be filtered by ir.rule. this is a refinement domain to see a subset of available records

Reply:
Yes, but such a refinement domain only needs to include a clause on 'model' and nothing else.